From f8911f0014dd0e1eb21dce3d113fcb1c1d225cb7 Mon Sep 17 00:00:00 2001
From: Brian Bennett <brian.bennett@joyent.com>
Date: Fri, 9 Sep 2016 14:35:35 -0700
Subject: [PATCH] MANTA-2986 marlin-agent update documentation should say to
 use updates-imgadm Reviewed by: Dave Pacheco <dap@joyent.com>

---
 docs/operator-guide/index.md | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/docs/operator-guide/index.md b/docs/operator-guide/index.md
index cf65691..6c9e4b2 100644
--- a/docs/operator-guide/index.md
+++ b/docs/operator-guide/index.md
@@ -1190,25 +1190,28 @@ Run this in each datacenter:
 
 Run this procedure for each datacenter whose Marlin agents you want to upgrade.
 
-1. Find the build you want to use.  If you have access to the builds under
-   /Joyent_Dev, you can run this from any machine with internet access and the
-   Manta CLI tools:
+1. Find the build you want to use using `updates-imgadm` in the global-zone
+   of the headnode.
 
-        headnode$ mfind -n 'marlin-master-.*\.tar.\gz' \
-            $(mget /Joyent_Dev/public/builds/marlin/master-latest)/marlin
+        headnode$ updates-imgadm list name=marlin
 
-2. Fetch the desired marlin-master-.\*.tar.gz tarball to /var/tmp on the
-   headnode.  We'll call that file's name TARBALL.
+2. Fetch the desired tarball to /var/tmp on the headnode.  The file will be
+   named `<UUID>-file.gz`.
+
+        headnode$ uuid=<UUID>
+        headnode$ cd /var/tmp
+        headnode$ updates-imgadm get=file -O "$uuid"
 
 3. Copy the tarball to each of the storage nodes:
 
         headnode$ sdc-oneachnode -n $(manta-adm cn -n storage) \
-            -d /var/tmp -g /var/tmp/TARBALL
+            -d /var/tmp -g "/var/tmp/${uuid}-file.gz"
 
 4. Apply the update to all shrimps with:
 
         headnode$ sdc-oneachnode -n $(manta-adm cn -n storage) \
-            '/opt/smartdc/agents/bin/apm install /var/tmp/TARBALL && svcs marlin-agent'
+            "/opt/smartdc/agents/bin/apm install /var/tmp/${uuid}-file.gz \
+            && svcs marlin-agent"
 
 5. Verify that agents are online with:
 
-- 
2.21.0

